### ISSUE TITLE
Bug: The token value of an escaped identifier differs with espree

### ISSUE THREAD

**Issue by JLHwung** [typescript-eslint/typescript-eslint#11036]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### Playground Link
https://typescript-eslint.io/play#ts=5.6.2&showAST=es&fileType=.tsx&code=DoVwDGBsCMQ&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=true

### Repro Code

```TypeScript
\u0061
```

### ESLint Config

```javascript

```

### tsconfig

```jsonc

```

### Expected Result

`.ast.tokens[0].value` should be `a` (Espree)

### Actual Result

`.ast.tokens[0].value` is `\\u0061`

### Additional Info

Whether a token value should be the escaped text / the raw text is certainly up to debate. If the goal is to align with espree, then I think we should use the escaped text.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `8.29.1` |

**Comment by bradzacher** [comment]
Okay so we chatted about this internally.

Acorn, babel and other parsers all "correctly" produce a token with `\\u0061` here because they are parsers that tokenise the source code themselves.

However we work on top of TS's parser -- meaning we're kind of stuck with TS's tokens and we cannot produce our own.

The only way for us to "correct" this behaviour is if we check every single token `Identifier` token to see if the underlying source code includes `\u` and if it does, run a regex to check for correctness and rewrite the token's value if it matches. This obviously is not a cheap operation to do.

Given that realistically usage of `\u` identifiers is going to be near-zero (if not actually zero) in any codebase AND desirability to lint `\u` identifiers is similarly going to be near-zero -- we do not think it is a good performance tradeoff for us to conform to this behaviour.

So we're going to close this as a "known and intentional divergence".
